src/GafferScene/BranchCreator.cpp

```diff
@@ -131,6 +131,52 @@ void validateDestination( const ScenePlug::ScenePath &destination )
 	}
 }
 
+IECore::CompoundObjectPtr attributesAfterCommonRoot(
+	const ScenePlug::ScenePath &queryPath, const ScenePlug::ScenePath &refPath, const ScenePlug *inPlug
+)
+{
+	unsigned int matchingLength = std::min( queryPath.size(), refPath.size() );
+	for( unsigned int i = 0; i < matchingLength; i++ )
+	{
+		if( queryPath[ i ] != refPath[ i ] )
+		{
+			matchingLength = i;
+			break;
+		}
+	}
+
+	IECore::CompoundObjectPtr result;
+
+	ScenePlug::ScenePath curPath = queryPath;
+	curPath.resize( matchingLength + 1 );
+	for( unsigned int i = matchingLength; i < queryPath.size(); i++ )
+	{
+		IECore::ConstCompoundObjectPtr curAttributes = inPlug->attributes( curPath );
```

Comment:
This makes a temporary context internally - might be worth reusing our own PathScope externally and using `attributesPlug()->getValue()` instead?